firebase-query.html

```diff
@@ -331,6 +334,9 @@
             this.async(function() {
               this.syncToMemory(function() {
                 this.splice('data', this.__indexFromKey(key), 1);
+                if (this.data.length === 0) {
```

Comment:
Thanks, 
http://stackoverflow.com/questions/15408416/how-to-handle-empty-arrays-in-firebase : "there is no way to force empty array into firebase".

And yes, data in memory does not always reflect what is in Firebase - see discussion #177 (zeroValue).